ansible/configs/osp-stf/post_software.yml

```diff
@@ -0,0 +1,195 @@
+  tags:
+  - osp-student-user-access
+
+- name: Providing the OSP standlone with flavors, test image, external network and tenant network
+  hosts: osps
+  gather_facts: false
+  tags:
+    - openstack-demo-environment
+  tasks:
+    - name: "Create flavors"
+      os_nova_flavor: 
+        cloud: standalone
+        name: "{{ item.name }}"
+        ram: "{{ item.ram }}"
+        vcpus: "{{ item.vcpus }}"
+        disk: "{{ item.disk }}"
+        ephemeral: "{{ item.ephemeral }}"
+        flavorid: "{{ item.flavorid }}"
+      loop:
+        - name: 'm1.small'
+          ram: '512'
+          vcpus: '1'
+          disk: '0'
+          ephemeral: '1'
+          flavorid: '1'
+        - name: 'm1.medium'
+          ram: '1024'
+          vcpus: '1'
+          disk: '0'
+          ephemeral: '0'
+          flavorid: '2'
+        
+    - name: Get cirros images
+      get_url: 
+        url: "http://download.cirros-cloud.net/0.5.1/cirros-0.5.1-x86_64-disk.img"
+        dest: "/home/{{ ansible_user }}/"
+
+    - name: Upload images
+      os_image:
+        cloud: standalone
+        name: 'cirros'
+        container_format: bare
+        disk_format: qcow2
+        is_public: yes
+        filename: '/home/{{ ansible_user }}/cirros-0.5.1-x86_64-disk.img'
+
+    - name: create tenant network
+      os_network:
+        cloud: standalone
+        name: test-network
+        external: false
+        wait: yes
+
+    - name: create tenant subnet
+      os_subnet:
+        cloud: standalone
+        network_name: test-network
+        name: test-subnet
+        cidr: 192.168.1.0/24
+        dns_nameservers:
+          - 8.8.8.8
+        wait: yes
+
+    - name: create provider networks and subnetworks
+      block:
+        - name: create provider nets
+          os_network:
+            cloud: standalone
+            name: external
+            external: True
+            shared: True
+            provider_network_type: flat
+            provider_physical_network: datacentre
+
+    # We are using the end range of our management network as a floating IP/provider FLAT network for instances.
+
+        - name: create provider subnets
+          os_subnet:
+            cloud: standalone
+            network_name: external
+            name: subnet-external
+            ip_version: 4
+            cidr: "{{ networks[0].subnet_cidr }}"
+            gateway_ip: "{{ networks[0].gateway_ip }}"
+            allocation_pool_start: "{{ networks[0].allocation_end }}"
+            allocation_pool_end: "{{ networks[0].subnet_cidr | ansible.netcommon.ipaddr('-2') | ansible.netcommon.ipaddr('address') }}"
+            enable_dhcp: yes
+            dns_nameservers:
+              - 8.8.8.8
+
+        - name: create a router
+          os_router:
+            cloud: standalone
+            name: test-router
+            network: external
+            interfaces:
+              - test-subnet
+      when: 'create_demo_provider_network | bool'
```

Comment:
Please put `when` statements of blocks before the `block`. It is difficult to match the `when` condition to the name when written in this manner. Rather use:

```
    - name: create provider networks and subnetworks
      when: 'create_demo_provider_network | bool'
      block:
```